code-challenges.js

```diff
@@ -3,61 +3,127 @@
+		// otherwise return '(argument) is not divisible by three'
+		return `${num} is not divisible by three`
+	}
+}
+
+// call the function
+console.log('Problem #1')
+console.log(divisibleByThree(num1))
+console.log(divisibleByThree(num2))
+console.log(divisibleByThree(num3))
 
 // --------------------2) Create a function that takes in the randomNouns variable and returns an array with all the words capitalized.
-
-var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
 
-
-
-
+let randomNouns = [
+	'streetlamp',
+	'potato',
+	'teeth',
+	'conclusion',
+	'nephew',
+	'temperature',
+	'database',
+]
+
+// create a function that takes an argument of an array of strings
+const capitalizeArray = arr => {
+	// iterate through array argument and capitalize each string
+	// create new variable to hold the new array
+	let newCapitalizedArray = arr.map(item => {
+		// isolate first character in each word to capitalize then add the rest of the word back on
+		return item.charAt(0).toUpperCase() + item.substr(1)
+	})
+	// return the new array with all words in the array argument capitalized
+	return newCapitalizedArray
+}
+
+// call the function
+console.log('Problem #2')
+console.log(capitalizeArray(randomNouns))
 
 // --------------------3) Create a function that takes in the mixedDataArray array and returns an array with ONLY NUMBERS sorted from least to greatest.
-
-var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"]
 // Expected output: [-8, 0, 8, 46, 59, 90, 107]
 
+let mixedDataArray = [true, 8, 'hello', 90, -8, null, 0, 46, 59, 107, 'hey!']
 
+// create a function that accepts an array of mixed data types
+const numbersOnlyArray = arr => {
+	// iterate through array argument separating out the number data types
+	// create a variable to hold the new array containing only numbers
+	let newNumbersArray = arr.filter(item => {
+		return typeof item === 'number'
+	})
+	// sort new number array (compare numbers to make sure values compared not just first digit)
+	// return a new array with only numbers sorted from least to greatest
+	return newNumbersArray.sort((a, b) => a - b)
+}
 
-
+// call the function
+console.log('Problem #3')
+console.log(numbersOnlyArray(mixedDataArray))
 
 // --------------------4) Create a function that takes in a string and logs the index of the first vowel.
 
-var vowelTester1 = "learn"
-// Expected output: 1
-var vowelTester2 = "throw"
-// Expected output: 3
-
-
+let vowelTester1 = 'learn' // Expected output: 1
+let vowelTester2 = 'throw' // Expected output: 3
 
+// create a varible that contains an array of vowels to compare against
 
+// create a function that takes a string as an argument
+const returnFirstVowel = str => {
+	// create a variable to hold the resulting search
+	let firstVowelResult = str.search(/[a, e, i, o, u, A, E, I, O, U]/g)
```

Comment:
Ooh! Regex! When using Regex, let's make sure we comment out exactly how its working in the line above it. Additionally, Regex allows you to utilize a flag to check for both uppercase and lowercase at the same time, to keep your code DRY.

Reply:
Yeah. I realized I should have written a note about that. I learned a fair amount about it because I didn't want to use something just because I found it. :)